### PR TITLE
DropSchema related changes for issue #1595

### PR DESCRIPTION
**Pull request by lmsurpre**
Previously, if `--drop-schema` was used without `--drop-schema-fhir`, `--drop-schema-oauth`, or `--drop-schema-jbatch` then it completed with success without actually doing anything.

I think what happened is that we were debating two different options for these args:
A. passing some kind of argument to `--drop-schema`, so it might be like `--drop-schema fhir` or `--drop-schema oauth jbatch`; or
B. using explicit flags for dropping the fhir, schema, or oauth tables, so usage would be like `--drop-schema-fhir --drop-schema-oauth`

But somehow we ended up with a combination of the two which I hope we can all agree is not very good.
This pull request modifies the current behavior and turns it into option B, removing support for the `--drop-schema` action.

Note: this PR does *not* actually address issue #1595 ...it only reworks the user experience of the cli so that others can avoid the error that I made initially.

### CODE REVIEW

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ public boolean isDuplicateSchema(SQLException x) {
 
     @Override
     public boolean isUndefinedName(SQLException x) {
-        return "42X05".equals(x.getSQLState());
```
Comment (prb112) [inline fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java:116]: Maybe this should be an OR with 

```
42883 | undefined_function
42P01 | undefined_table
42P02 | undefined_parameter
42704 | undefined_object
```

File: fhir-persistence-schema/README.md [hunk 2]
```diff
@@ -85,7 +85,7 @@ The following sections include common values for `OPTIONS`.
 ```
 --prop-file db2.properties
 --schema-name FHIRDATA
---drop-schema
+--drop-schema-fhir
 --drop-admin
 --confirm-drop
 ```
```
Comment (prb112) [inline fhir-persistence-schema/README.md:91]: maybe we should add all three in this example, or mention in a comment below this update that you can use --drop-schema-oauth and jbatch?

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java [hunk 3]
```diff
@@ -424,7 +423,7 @@ protected void dropSchema() {
                     JdbcTarget target = new JdbcTarget(c);
                     IDatabaseAdapter adapter = getDbAdapter(dbType, target);
 
-                    if (this.dropSchema) {
+                    if (dropFhirSchema || dropOauthSchema || dropJavaBatchSchema) {
```
Comment (prb112) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:426]: is the adminDrop implicit here? what if we just call dropAdmin

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java [hunk 4]
```diff
@@ -1387,20 +1385,13 @@ protected void process() {
 
         if (addKeyForTenant != null) {
             addTenantKey();
-        } else if (this.dropSchema) {
+        } else if (this.dropAdmin || this.dropFhirSchema || this.dropJavaBatchSchema || this.dropOauthSchema) {
```
Comment (prb112) [inline fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:1388]: See 426

**Review by prb112 — CHANGES_REQUESTED** [review]
Left comments that I want to confirm before LGTM

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java (hunk 3 above)
Comment (lmsurpre) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:426]: dropAdmin is covered separately in the if case just below this

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java:116]: empirically I found the `42704` is what we got back for undefined functions, tables, and indexes.  not sure how to reconcile that with the provided list, but its easy-enough for me to or them

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java (hunk 3 above)
Comment (punktilious) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:426]: I think we should (later) revisit the whole option control logic in this Main...too much entropy.

File: fhir-persistence-schema/README.md (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence-schema/README.md:91]: sure, I can add the comment

File: fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java (hunk 3 above)
Comment (prb112) [reply fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/Main.java:426]: +1 on Robin's comment.  it's definitely growing

**Review by punktilious — APPROVED** [review]
LGTM, subject to Paul's comments.

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java (hunk 1 above)
Comment (prb112) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java:116]: I wasn't sure if we should or not, it was more of a check-in, not necessarily that we had to or-then

**Review by prb112 — APPROVED** [review]

File: fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgresql/PostgreSqlTranslator.java:116]: ok, i went ahead and added them

File: fhir-persistence-schema/README.md (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence-schema/README.md]: ended up adding two different subsections; one for dropping just the fhir schema and one for dropping them all

**Pull request merged**